plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/RestJobManagement.groovy

```diff
@@ -351,6 +353,21 @@ class RestJobManagement extends AbstractJobManagement implements DeferredJobMana
         }
 
         if (isXmlDifferent(existingXml, item.xml)) {
+            if (item instanceof Folder) {
+                /*
+                Folders can contain credentials scoped to the items they contain. The credentials are
+                stored in the folder XML and not in the Jenkins configuration files, so we copy over any
+                manually set properties from the existing folders.
+                Approach suggested in https://issues.jenkins-ci.org/browse/JENKINS-44681
+                 */
+                def existingXmlParsed = new XmlParser().parseText(existingXml)
+                if (existingXmlParsed.properties != null) {
+                    def newXmlParsed = new XmlParser().parseText(item.xml)
+                    Node node = existingXmlParsed.properties[0] as Node
+                    newXmlParsed.properties[0].replaceNode(node)
```

Comment:
Shouldn't we add the node only if there is no node in `newXmlParsed`. You could configure credentials using the Job DSL configure closure which would not be possible anymore with this approach (at least it could be changed after the first creation).

Reply:
Agree on:

> Shouldn't we add the node only if there is no node in newXmlParsed.

I'll do this change. 

Regarding 

> Also is it safe to assume we have to take and replace the first node from the properties array?

I'm taking the whole properties node, so it not only copies all the credentials but also there kind of properties. Think this makes sense? It's a more general solution than the one proposed in the Jenkins issue tracker.

Reply:
I don't know what other configuration can be in the properties node. If we replace all nodes it should at least be mentioned in the readme, that existing properties will be used in case the generated DSL does not contain any properties.

Reply:
I see your point. Sounds like we should look up for the credentials property instead, it's a safer approach, specially if add the validation for not replacing existing configurations provided by the scripts.